blog/posts/making-the-remix-austin-blog.mdx

```diff
@@ -0,0 +1,214 @@
+
+## We got the goods. Now what?
+
+The solution technically worked, but it was lacking quite a bit. The first being understanding.
+I'm no fan of "copy and paste" into production code. Sure, publicly available solutions are great
+starting points. But you shouldn't put any code into a codebase that you don't understand.
+
+I took a good long look at what I copied and got my head around how it worked. The basic idea was this:
+there's a NodeJS script that takes all of the `.mdx` files and any of their dependencies
+and compiles them using `mdx-bundler`. The final result is a JSON array, with each post as an
+object in that array. Those objects have some frontmatter and a raw source string for a React
+component. That raw React component is the full content of the blog post.
+
+The array then gets written into a file called `post-cache.json` in our `/public`
+directory, making it available to all loaders in the site. Loaders for the blog list route and the
+individual blog post route make a fetch for the file, pick out the data they need, and pass the
+raw React component to a helper function from `mdx-bundler` which converts into a regular
+React component, and finally we display it in our page like good ol' JSX.
+
+And lastly, all `.mdx` files are watched by Node when we run this script in development mode. If any
+of their content changes, `post-cache.json` is regenerated with the updated contents.
+
+<BlogImage
+  src={mattConfused}
+  alt="Not confused at all"
+  caption="Confused like my face here? Just reread that section a few more times."
+  maxWidth={300}
+/>
+
+So this is pretty good, right? But it could be better.
+
+For starters, the entire `post-cache.json` file was being regenerated in dev mode for any
+single change to a blog post. That means every blog was getting built and compiled even if I'm only
+focused on editing one post. That's a lot of wasted power. Ideally, I would like for my app
+to compile the one post I'm working on at the moment.
+
+Secondly, the `post-cache.json` file itself was capable of getting quite large quite fast. There's **a lot** of
+boilerplate that goes into that raw React component for each compiled post. Imagine duplicating that boilerplate
+for every single post! It quickly bloats in size. Did we really need to have an array with ALL of the blog
+information in it (front matter and content)? It seemed better to have an array that just contained
+each post's front matter. That would then allow us to list all of our blog posts in the `/blog` page,
+without having to worry about all that boilerplate.
+
+Lastly, if each compiled blog post has a lot of boilerplate, it seemed like a better idea to have each
+blog post built on demand by the server when the post is requested. That would save us from having one 
+mega-file with everything in it. There would be a tradeoff because we couldn't cache that mega-file, but `mdx-bundler` is quite fast.
```

Comment:
Not for this post/PR, but I would like to look into returning cache headers from the loaders for each blog, that way we don't have to run mdx-bundler each time the blog is requested

> There would be a tradeoff because we couldn't cache that mega-file

Reply:
I think I'll just remove that last sentence. Using cache headers to cache the post would make it a moot point.